### PR TITLE
Change 'SSH' mentions to 'Script' for Ansible

### PR DESCRIPTION
**Pull request by sabuchan07**
* [ x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [ x] Foreman 3.3/Katello 4.5
* [ ] Foreman 3.2/Katello 4.4
* [ ] Foreman 3.1/Katello 4.3
* For Foreman 3.0 or older, please create a separate PR.
* We do not accept PRs for Foreman 2.3 or older.

### CODE REVIEW

File: guides/common/modules/con_about-running-jobs-on-hosts.adoc [hunk 1]
```diff
@@ -8,7 +8,7 @@ For custom Ansible roles that you create, or roles that you download, you must i
 Before you can use Ansible roles, you must import the roles into {Project} from the {SmartProxy} where they are installed.
 
 Communication occurs through {SmartProxyServer}, which means that {ProjectServer} does not require direct access to the target host, and can scale to manage many hosts.
-Remote execution uses the SSH service that must be enabled and running on the target host.
+Remote execution uses the Script service that must be enabled and running on the target host.
```
Comment (adamruzicka) [inline guides/common/modules/con_about-running-jobs-on-hosts.adoc:11]: No matter how we call it in Foreman/Satellite, on it still does not change a name of the service running on the target host. On the target host it is still a ssh service (usually called sshd and being an ssh server from openssh). All in all this one should probably stay the way it was

File: guides/common/modules/proc_configuring-a-kickstart-template-to-distribute-ssh-keys-during-provisioning.adoc [hunk 2]
```diff
@@ -1,14 +1,14 @@
-[id="configuring-a-kickstart-template-to-distribute-ssh-keys-during-provisioning_{context}"]
-= Configuring a Kickstart Template to Distribute SSH Keys during Provisioning
+[id="configuring-a-kickstart-template-to-distribute-script-keys-during-provisioning_{context}"]
+= Configuring a Kickstart Template to Distribute Script Keys during Provisioning
```
Comment (adamruzicka) [inline guides/common/modules/proc_configuring-a-kickstart-template-to-distribute-ssh-keys-during-provisioning.adoc:2]: Similarly, the keys are specific to the transport mechanism (SSH) so it should stay the way it was. The same goes for all the other references to keys in this doc

File: guides/common/modules/proc_configuring-a-kickstart-template-to-distribute-ssh-keys-during-provisioning.adoc [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
 
-You can add a `remote_execution_ssh_keys` snippet to your custom kickstart template to deploy SSH Keys to hosts during provisioning.
+You can add a `remote_execution_script_keys` snippet to your custom kickstart template to deploy Script Keys to hosts during provisioning.
```
Comment (adamruzicka) [inline guides/common/modules/proc_configuring-a-kickstart-template-to-distribute-ssh-keys-during-provisioning.adoc:4]: We have not renamed the snippet

File: guides/common/modules/proc_configuring-kerberos-authentication-for-remote-execution.adoc [hunk 4]
```diff
@@ -1,7 +1,7 @@
 [id="configuring-kerberos-authentication-for-remote-execution_{context}"]
 = Configuring Kerberos Authentication for Remote Execution
 
-You can use Kerberos authentication to establish an SSH connection for remote execution on {Project} hosts.
+You can use Kerberos authentication to establish a Script connection for remote execution on {Project} hosts.
```
Comment (adamruzicka) [inline guides/common/modules/proc_configuring-kerberos-authentication-for-remote-execution.adoc:4]: All the references to the underlying technology should stay. Even if you're running a script, the connection itself is ssh

File: guides/common/modules/proc_executing-a-remote-job.adoc [hunk 5]
```diff
@@ -21,7 +21,7 @@ See {ManagingHostsDocURL}creating-a-job-template_managing-hosts[Creating a Job T
 . Optional: To configure advanced settings for the job, click *Display advanced fields*.
 Some of the advanced settings depend on the job template, the following settings are general:
 
-* *Effective user* defines the user for executing the job, by default it is the SSH user.
+* *Effective user* defines the user for executing the job, by default it is the Script user.
```
Comment (adamruzicka) [inline guides/common/modules/proc_executing-a-remote-job.adoc:24]: I'm not sure how to handle this, but there are scenarios in which this is not true. When ssh is used as a transport this holds, but when the pull provider is used, then the default user is `root`

File: guides/common/modules/proc_using-the-api-to-obtain-ssh-keys-for-remote-execution.adoc [hunk 6]
```diff
@@ -1,31 +1,31 @@
...
-= Using the {Project} API to Obtain SSH Keys for Remote Execution
+[id="using-the-api-to-obtain-script-keys-for-remote-execution_{context}"]
+= Using the {Project} API to Obtain Script Keys for Remote Execution
 
 To use the {Project} API to download the public key from {SmartProxy}, complete this procedure on each target host.
 
 .Procedure
 
-. On the target host, create the `~/.ssh` directory to store the SSH key:
+. On the target host, create the `~/.script` directory to store the Script key:
```
Comment (adamruzicka) [inline guides/common/modules/proc_using-the-api-to-obtain-ssh-keys-for-remote-execution.adoc:8]: The path stays

**Review by adamruzicka — CHANGES_REQUESTED** [review]

File: guides/common/modules/con_about-running-jobs-on-hosts.adoc (hunk 1 above)
Comment (sabuchan07) [reply guides/common/modules/con_about-running-jobs-on-hosts.adoc:11]: Hi, @adamruzicka !  Thanks for your review.  The ticket attached to this PR says "The ssh proxy feature will be renamed to script. All mentions of ssh in the installer will be renamed to script."

If these changes are all incorrect, what exactly is it that's being changed?  I'm going to have another PR for the same ticket/different guide, and I want to make sure that one is done correctly before submitting.  Thanks!

File: guides/common/modules/proc_assigning-ansible-roles-to-an-existing-host.adoc [hunk 7]
```diff
@@ -21,7 +21,7 @@ Use the *<* button to remove a role, or the *<<* button to remove all roles.
 . Click *Confirm*.
 
 After you assign Ansible roles to hosts, you can use Ansible for remote execution.
-For more information, see {ConfiguringAnsibleDocURL}ssh-keys-for-remote-execution-overview_ansible[Distributing SSH Keys for Remote Execution].
+For more information, see https://docs.theforeman.org/nightly/Configuring_Ansible/index-foreman-el.html#ssh-keys-for-remote-execution-overview_ansible[Distributing SSH Keys for Remote Execution].
```
Comment (maximiliankolb) [inline guides/common/modules/proc_assigning-ansible-roles-to-an-existing-host.adoc:24]: Please revert this change.

File: guides/common/modules/proc_configuring-your-deployment-to-run-ansible-roles.adoc [hunk 8]
```diff
@@ -35,8 +35,8 @@ If you want to use custom or third party Ansible roles, ensure to configure an e
 --enable-foreman-proxy-plugin-ansible
 ----
 
-. Distribute SSH keys to enable {SmartProxies} to connect to hosts using SSH.
-For more information, see {ManagingHostsDocURL}ssh-keys-for-remote-execution-overview_managing-hosts[Distributing SSH Keys for Remote Execution] in the _Managing Hosts_ guide.
+. Distribute Script keys to enable {SmartProxies} to connect to hosts using Script.
+For more information, see {ManagingHostsDocURL}ssh-keys-for-remote-execution-overview_managing-hosts[Distributing Script Keys for Remote Execution] in the _Managing Hosts_ guide.
```
Comment (maximiliankolb) [inline guides/common/modules/proc_configuring-your-deployment-to-run-ansible-roles.adoc:39]:
```suggestion
For more information, see {ManagingHostsDocURL}ssh-keys-for-remote-execution-overview_managing-hosts[Distributing Script Keys for Remote Execution] in _{ManagingHostsDocTitle}_.
```

I assume SSH keys remain SSH keys, so you probably have to change this as well. If you do, you can simply checkout the file from the master branch and ignore my suggestion.

File: guides/common/modules/proc_distributing-keys-for-remote-execution.adoc [hunk 9]
```diff
@@ -1,19 +1,19 @@
...
+Use one of the following methods to distribute the public Script key from {SmartProxy} to target hosts:
 
-. xref:distributing-ssh-keys-for-remote-execution-manually_{context}[].
-. xref:using-the-api-to-obtain-ssh-keys-for-remote-execution_{context}[].
-. xref:configuring-a-kickstart-template-to-distribute-ssh-keys-during-provisioning_{context}[].
-. For new {Project} hosts, you can deploy SSH keys to {Project} hosts during registration using the global registration template.
+. https://docs.theforeman.org/nightly/Managing_Hosts/index-foreman-el.html#distributing-ssh-keys-for-remote-execution-manually_managing-hosts[Distributing Script Keys for Remote Execution Manually].
+. https://docs.theforeman.org/nightly/Configuring_Ansible/index-foreman-el.html#using-the-api-to-obtain-ssh-keys-for-remote-execution_ansible[Using the Foreman API to Obtain Script Keys for Remote Execution].
+. https://docs.theforeman.org/nightly/Configuring_Ansible/index-foreman-el.html#configuring-a-kickstart-template-to-distribute-ssh-keys-during-provisioning_ansible[Configuring a Kickstart Template to Distribute Script Keys during Provisioning].
```
Comment (maximiliankolb) [inline guides/common/modules/proc_distributing-keys-for-remote-execution.adoc:12]: Please do not use links like this.

File: guides/common/modules/proc_managing-and-monitoring-hosts-using-cockpit.adoc [hunk 10]
```diff
@@ -14,7 +14,7 @@ ifdef::satellite[]
 ** For {RHEL} 7, see https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/managing_systems_using_the_rhel_7_web_console/getting-started-with-the-rhel-web-console_system-management-using-the-rhel-7-web-console#installing-the-web-console_getting-started-with-the-web-console[Installing the web console] in the _Managing systems using the RHEL 7 web console_ guide.
 endif::[]
 * {Project} or {SmartProxy} can authenticate to the host with SSH keys.
-For more information, xref:Distributing_SSH_Keys_for_Remote_Execution_{context}[].
+For more information, see https://docs.theforeman.org/nightly/Managing_Hosts/index-foreman-el.html#Distributing_SSH_Keys_for_Remote_Execution_managing-hosts[Distributing_SSH_Keys_for_Remote_Execution].
```
Comment (maximiliankolb) [inline guides/common/modules/proc_managing-and-monitoring-hosts-using-cockpit.adoc:17]: same

File: guides/common/modules/proc_migrating-from-katello-agent-to-remote-execution.adoc [hunk 11]
```diff
@@ -36,7 +36,7 @@ WARNING: If your host is installed on {oVirt} version 4.4 or lower, do not remov
 # {package-remove} katello-agent
 ----
 . Distribute the remote execution SSH keys to the content hosts.
-For more information, see xref:Distributing_SSH_Keys_for_Remote_Execution_{context}[].
+For more information, see https://docs.theforeman.org/nightly/Managing_Hosts/index-foreman-el.html#Distributing_SSH_Keys_for_Remote_Execution_managing-hosts[Distributing Script Keys for Remote Execution].
```
Comment (maximiliankolb) [inline guides/common/modules/proc_migrating-from-katello-agent-to-remote-execution.adoc:39]: same

File: guides/common/modules/proc_upgrading-rhel7-hosts-to-rhel8.adoc [hunk 12]
```diff
@@ -15,7 +15,7 @@ endif::[]
 * Enable remote execution feature on {Project}.
 For more information, see xref:Configuring_and_Setting_Up_Remote_Jobs_{context}[].
 * Distribute {Project} SSH keys to the hosts that you want to upgrade.
-For more information, see xref:Distributing_SSH_Keys_for_Remote_Execution_{context}[].
+For more information, see https://docs.theforeman.org/nightly/Managing_Hosts/index-foreman-el.html#Distributing_SSH_Keys_for_Remote_Execution_managing-hosts[Distributing Script Keys for Remote Execution].
```
Comment (maximiliankolb) [inline guides/common/modules/proc_upgrading-rhel7-hosts-to-rhel8.adoc:18]: same

File: guides/common/modules/proc_assigning-ansible-roles-to-an-existing-host.adoc (hunk 7 above)
Comment (sabuchan07) [reply guides/common/modules/proc_assigning-ansible-roles-to-an-existing-host.adoc:24]: Hi, @maximiliankolb !  The original links weren't passing the PR checks, and when I changed them, the checks all passed.  Any idea why this might be the case?

Comment (maximiliankolb) [reply guides/common/modules/proc_assigning-ansible-roles-to-an-existing-host.adoc:24]: That sounds strange, yes. But because the pipeline for "master" branch build, your feature branch without any change in this file and without change to the anchor itself should definitely work!

**Comment by sabuchan07** [comment]
@maximiliankolb , I changed the links back to how they were in the original document...and now the checks are failing again.

File: guides/common/modules/con_about-running-jobs-on-hosts.adoc (hunk 1 above)
Comment (adamruzicka) [reply guides/common/modules/con_about-running-jobs-on-hosts.adoc]: > All mentions of ssh in the installer will be renamed to script

The installer had bunch of options looking like  `--foreman-proxy-plugin-remote-execution-ssh-$something`, those will now be `--foreman-proxy-plugin-remote-execution-script-$something`.

> The ssh proxy feature will be renamed to script.

This, as seen in Infrastructure > Capsules.

> what exactly is it that's being changed?

In addition to the two points above:
- provider type (as seen in Hosts > Job templates > $template > Job tab > Provider type)
- Names of default job templates will now have script instead of ssh in their names

And that should be it as far as I recall.

File: guides/common/modules/proc_assigning-ansible-roles-to-an-existing-host.adoc [hunk 13]
```diff
@@ -21,7 +21,7 @@ Use the *<* button to remove a role, or the *<<* button to remove all roles.
 . Click *Confirm*.
 
 After you assign Ansible roles to hosts, you can use Ansible for remote execution.
-For more information, see {ConfiguringAnsibleDocURL}ssh-keys-for-remote-execution-overview_ansible[Distributing SSH Keys for Remote Execution].
+For more information, see {ConfiguringAnsibleDocURL}ssh-keys-for-remote-execution-overview_ansible[Distributing SSH Keys for Remote Execution]
```
Comment (maximiliankolb) [inline guides/common/modules/proc_assigning-ansible-roles-to-an-existing-host.adoc:24]:
```suggestion
For more information, see {ConfiguringAnsibleDocURL}ssh-keys-for-remote-execution-overview_ansible[Distributing SSH Keys for Remote Execution].
```

File: guides/common/modules/proc_distributing-keys-for-remote-execution-manually.adoc [hunk 14]
```diff
@@ -1,19 +1,19 @@
-[id="distributing-ssh-keys-for-remote-execution-manually_{context}"]
-= Distributing SSH Keys for Remote Execution Manually
+[id="distributing-script-keys-for-remote-execution-manually_{context}"]
+= Distributing Script Keys for Remote Execution Manually
 
-To distribute SSH keys manually, complete the following steps:
+To distribute Script keys manually, complete the following steps:
```
Comment (maximiliankolb) [inline guides/common/modules/proc_distributing-keys-for-remote-execution-manually.adoc:4]: I am unsure if this is correct. cc @adamruzicka

File: guides/common/modules/proc_distributing-keys-for-remote-execution.adoc [hunk 15]
```diff
@@ -1,19 +1,19 @@
...
+To use SSH, you must distribute the public SSH key from {SmartProxy} to its attached hosts that you want to manage.
 Ensure that the SSH service is enabled and running on the hosts.
 Configure any network or host-based firewalls to enable access to port 22.
 
 Use one of the following methods to distribute the public SSH key from {SmartProxy} to target hosts:
 
 . xref:distributing-ssh-keys-for-remote-execution-manually_{context}[].
 . xref:using-the-api-to-obtain-ssh-keys-for-remote-execution_{context}[].
-. xref:configuring-a-kickstart-template-to-distribute-ssh-keys-during-provisioning_{context}[].
+. xref:configuring-a-kickstart-template-to-distribute-ssh-keys-during-provisioning_{context}[]
```
Comment (maximiliankolb) [inline guides/common/modules/proc_distributing-keys-for-remote-execution.adoc:12]:
```suggestion
. xref:configuring-a-kickstart-template-to-distribute-ssh-keys-during-provisioning_{context}[].
```

**Review by maximiliankolb — CHANGES_REQUESTED** [review]

File: guides/common/modules/proc_distributing-keys-for-remote-execution-manually.adoc (hunk 14 above)
Comment (adamruzicka) [reply guides/common/modules/proc_distributing-keys-for-remote-execution-manually.adoc:4]: It is not, there are no such things as script keys. I already pointed that out in another place and hoped it would be enough

**Comment by sabuchan07** [comment]
Hi, @ekohl !  I did a force push on this after rebasing (like we covered yesterday), and now my links are broken, even though these were the links in the original document.  (I looked at the log, and it just said they were not successful.). Could my rebasing have broken my links?

**Comment by sabuchan07** [comment]
@ekohl I did!  That's why I was confused when this happened again.

**Comment by sabuchan07** [comment]
Hi, @maximiliankolb !  This one should also be ready to merge.

File: guides/common/modules/proc_distributing-keys-for-remote-execution.adoc [hunk 16]
```diff
@@ -3,17 +3,20 @@
...
 
 Use one of the following methods to distribute the public SSH key from {SmartProxy} to target hosts:
 
 . xref:distributing-ssh-keys-for-remote-execution-manually_{context}[].
 . xref:using-the-api-to-obtain-ssh-keys-for-remote-execution_{context}[].
 . xref:configuring-a-kickstart-template-to-distribute-ssh-keys-during-provisioning_{context}[].
-. For new {Project} hosts, you can deploy SSH keys to {Project} hosts during registration using the global registration template.
+
+For new {Project} hosts, you can deploy SSH keys to {Project} hosts during registration using the global registration template.
+
```
Comment (maximiliankolb) [inline guides/common/modules/proc_distributing-keys-for-remote-execution.adoc:16]: I don't get why this should be changed.

File: guides/common/modules/proc_distributing-keys-for-remote-execution.adoc [hunk 17]
```diff
@@ -3,17 +3,20 @@
 
 To use SSH keys for authenticating remote execution connections, you must distribute the public SSH key from {SmartProxy} to its attached hosts that you want to manage.
 Ensure that the SSH service is enabled and running on the hosts.
+
```
Comment (maximiliankolb) [inline guides/common/modules/proc_distributing-keys-for-remote-execution.adoc:6]:
```suggestion
```

File: guides/common/modules/proc_configuring-your-deployment-to-run-ansible-roles.adoc [hunk 18]
```diff
@@ -48,7 +48,9 @@ If you want to use custom or third party Ansible roles, ensure to configure an e
 ----
 
 . Distribute SSH keys to enable {SmartProxies} to connect to hosts using SSH.
+
```
Comment (maximiliankolb) [inline guides/common/modules/proc_configuring-your-deployment-to-run-ansible-roles.adoc:51]:
```suggestion
```

File: guides/common/modules/proc_configuring-your-deployment-to-run-ansible-roles.adoc [hunk 19]
```diff
@@ -48,7 +48,9 @@ If you want to use custom or third party Ansible roles, ensure to configure an e
...
 For more information, see {ManagingHostsDocURL}ssh-keys-for-remote-execution-overview_managing-hosts[Distributing SSH Keys for Remote Execution] in _{ManagingHostsDocTitle}_.
+
```
Comment (maximiliankolb) [inline guides/common/modules/proc_configuring-your-deployment-to-run-ansible-roles.adoc:53]:
```suggestion
```

File: guides/common/modules/proc_assigning-ansible-roles-to-an-existing-host.adoc [hunk 20]
```diff
@@ -16,6 +16,8 @@ You can add more than one role.
 . Click *Submit*.
 
 After you assign Ansible roles to hosts, you can use Ansible for remote execution.
+For more information, see {ConfiguringAnsibleDocURL}ssh-keys-for-remote-execution-overview_ansible[Distributing SSH Keys for Remote Execution].
```
Comment (maximiliankolb) [inline guides/common/modules/proc_assigning-ansible-roles-to-an-existing-host.adoc:19]:
```suggestion
For more information, see {ConfiguringAnsibleDocURL}ssh-keys-for-remote-execution-overview_ansible[Distributing SSH Keys for Remote Execution] in _{ConfiguringAnsibleDocTitle}_.
```

**Review by maximiliankolb — COMMENTED** [review]
Please keep your PR as small as possible. If you want to add some extra new lines at places, that can be a valid commit, but this looks more like a mishap when trying to revert changes from "script" back to "ssh".

File: guides/common/modules/proc_distributing-keys-for-remote-execution.adoc (hunk 16 above)
Comment (sabuchan07) [reply guides/common/modules/proc_distributing-keys-for-remote-execution.adoc:16]: @maximiliankolb Can you please clarify this comment?

**Comment by sabuchan07** [comment]
@maximiliankolb Suggestions fixed

File: guides/common/modules/proc_distributing-keys-for-remote-execution.adoc [hunk 21]
```diff
@@ -11,9 +11,10 @@ Use one of the following methods to distribute the public SSH key from {SmartPro
 . xref:using-the-api-to-obtain-ssh-keys-for-remote-execution_{context}[].
 . xref:configuring-a-kickstart-template-to-distribute-ssh-keys-during-provisioning_{context}[].
 . For new {Project} hosts, you can deploy SSH keys to {Project} hosts during registration using the global registration template.
+
```
Comment (maximiliankolb) [inline guides/common/modules/proc_distributing-keys-for-remote-execution.adoc:14]:
```suggestion
```

**Comment by sabuchan07** [comment]
@maximiliankolb Change made

File: guides/common/modules/proc_configuring-kerberos-authentication-for-remote-execution.adoc (hunk 4 above)
Comment (maximiliankolb) [reply guides/common/modules/proc_configuring-kerberos-authentication-for-remote-execution.adoc:4]: This is still unresolved.

**Review by maximiliankolb — CHANGES_REQUESTED** [review]

File: guides/common/modules/proc_assigning-ansible-roles-to-an-existing-host.adoc [hunk 22]
```diff
@@ -16,6 +16,8 @@ You can add more than one role.
 . Click *Submit*.
 
 After you assign Ansible roles to hosts, you can use Ansible for remote execution.
+For more information, see {ConfiguringAnsibleDocURL}ssh-keys-for-remote-execution-overview_ansible[Distributing SSH Keys for Remote Execution] in _{ConfiguringAnsibleDocTitle}_.
+
```
Comment (maximiliankolb) [inline guides/common/modules/proc_assigning-ansible-roles-to-an-existing-host.adoc:20]:
```suggestion
```

It seems that this link is already present in line 21.

File: guides/common/modules/proc_assigning-ansible-roles-to-an-existing-host.adoc [hunk 23]
```diff
@@ -16,6 +16,7 @@ You can add more than one role.
 . Click *Submit*.
 
 After you assign Ansible roles to hosts, you can use Ansible for remote execution.
+
```
Comment (maximiliankolb) [inline guides/common/modules/proc_assigning-ansible-roles-to-an-existing-host.adoc:19]:
```suggestion
```

**Review by maximiliankolb — APPROVED** [review]
LGTM.

**Comment by maximiliankolb** [comment]
Merged to master and cherry-picked: ✅ 
   b2a1bd529..7dd68e339  3.4 -> 3.4
   6c9bb20c7..fb1eb9f14  3.3 -> 3.3
